### PR TITLE
Correct error message for aten::_local_scalar_dense on meta tensor

### LINKED ISSUE #119588: Bad error message for aten::_local_scalar_dense on meta tensor
### 🐛 Describe the bug

```
NotImplementedError: aten::_local_scalar_dense: attempted to run this operator with Meta tensors, but there was no abstract impl or Meta kernel registered. You may have run into this message while using an operator with PT2 compilation APIs (torch.compile/torch.export); in order to use this operator with those APIs you'll need to add an abstract impl.Please see the following doc for next steps: https://docs.google.com/document/d/1_W62p8WJOQQUzPsJYa7s701JXt0qf2OfLub2sbkHOaU/edit
```

This error message is bad because you can't actually define an abstract impl for this (or any other data dependent) operator.

cc @malfet @eellison @bdhirsh @msaroufim @anijain2305 @zou3519 @chauhang 

### Versions

main

### PR DESCRIPTION
**Pull request by Saadelkouari**
registering a meta for aten::_local_scalar_dense with a different error message.

Fixes pytorch#119588

cc @ezyang @zou3519

### CODE REVIEW

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -6237,6 +6237,15 @@ def meta_channel_shuffle(input, groups):
     )
 
 
+@register_meta(aten._local_scalar_dense)
+def meta_local_scalar_dense(self:Tensor):
+    torch._check(
+        not self.is_meta,
+        lambda: "aten::_local_scalar_dense operator (aka Tensor.item()) cannot be called on meta tensors.",
+    )
+    return torch.empty_like(self)
```
Comment (ezyang) [inline torch/_meta_registrations.py:6246]: Don't bother returning self.is_meta, just raise an assertion unconditionally. When something is "not meta but meta-ish" (aka fake tensor) there is a completely overridden path in fake tensor.

Can you add a test?

**Review by ezyang — CHANGES_REQUESTED** [review]
Thanks! Looks promising.

File: torch/_meta_registrations.py (hunk 1 above)
Comment (Saadelkouari) [reply torch/_meta_registrations.py:6246]: So I'll change the meta to the following:
```python
@register_meta(aten._local_scalar_dense)
def meta_local_scalar_dense(self:Tensor):
    assert not self.is_meta, "aten::_local_scalar_dense operator (aka Tensor.item()) cannot be called on meta tensors."  
    return torch.empty_like(self)
```

and I will add the following test to the class `TestMeta` in `test/test_meta.py`:
```python
    def test_local_scalar_dense_call(self):
        with self.assertRaisesRegex(RuntimeError, "aten::\_local\_scalar\_dense operator (aka Tensor\.item()) cannot be called on meta tensors\."):
            meta_tensor = torch.randn(1.0, device='meta')
            meta_tensor.item()
```
is this the right way to go about it?

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**